sda/protocol.go

```diff
@@ -37,59 +38,62 @@ type ProtocolInstance interface {
 	Shutdown() error
 }
 
-// NewProtocol is the function-signature needed to instantiate a new protocol
-type NewProtocol func(*TreeNodeInstance) (ProtocolInstance, error)
+var protocols = NewProtocolStorage()
 
-// ProtocolNameToID returns the ProtocolID corresponding to the given name
-func ProtocolNameToID(name string) ProtocolID {
-	url := network.NamespaceURL + "protocolname/" + name
-	return ProtocolID(uuid.NewV3(uuid.NamespaceURL, url))
+// ProtocolStorage holds all protocols either globally or per-Conode.
+type ProtocolStorage struct {
+	// Instantiators maps the name of the protocols to the `NewProtocol`-
+	// methods.
+	Instantiators map[string]NewProtocol
```

Comment:
It looks like the `ProtocolStorage` is only used internally (in the `sda` package). I also think it should be private (as well as `NewProtocolStorage`) if there isn't a good reason for it to be public.